intro-to-R-tidyverse/02-intro_to_ggplot2.Rmd

```diff
@@ -0,0 +1,306 @@
+2. `mapping`, which is a special list made with the `aes()` function to describe which values will be used for each **aes**thetic component of the plot, such as the x and y coordinates of each point. 
+(If you find calling things like the x and y coordinates "aesthetics" confusing, don't worry, you are not alone.)  
+
+```{r}
+ggplot(
+  tumor_normal_df, # This first argument is the data.frame with the data we want to plot
+  aes(
+    x = log_fold_change, # This is the column name of the values we want to use
+    # for the x coordinates
+    y = neg_log_p_val
+  ) # This is the column name of the data we want for the y-axis
+)
+```
+
+You'll notice this plot doesn't have anything on it because we haven't 
+specified a plot type yet. 
+To do that, we will add another ggplot layer with `+` which will specify exactly what we want to plot.
+A volcano plot is a special kind of scatter plot, so to make that we will want to plot individual points, which we can do with `geom_point()`.
+
+```{r}
+# This first part is the same as before
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val
+  )
+) +
+  # Now we are adding on a layer to specify what kind of plot we want
+  geom_point()
+```
+
+Here's a brief summary of ggplot2 structure. 
+[ggplot2 structure](diagrams/ggplot_structure.png)
+
+### Adjust our ggplot
+
+Now that we have a base plot that shows our data, we can add layers on to it and adjust it.
+We can adjust the color points using `color`. 
+
+```{r}
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val,
+    color = avg_expression
+  ) # We added this argument to color code the points!
+) +
+  geom_point()
+```
+
+Because we have so many points overlapping one another, we will want to adjust 
+the transparency, which we can do with an `alpha` argument. 
+
+```{r}
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val,
+    color = avg_expression
+  )
+) +
+  geom_point(alpha = 0.2) # We are using the `alpha` argument to make our points transparent
+```
+
+Notice that we added the alpha within the `geom_point()` function, not to the `aes()`. 
+We did this because we want all of the points to have the same level of transparency, and it will not vary depending on any variable in the data.
+We can also change the background and appearance of the plot as a whole by adding a `theme`.
+
+```{r}
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val,
+    color = avg_expression
+  )
+) +
+  geom_point(alpha = 0.2) +
+  theme_bw() # Add on this set of appearance presets to make it pretty
+```
+
+We can change the x and y labels by using `ylab` and `xlab` functions and add a
+title using `ggtitle`.
+
+```{r}
+ggplot(
+  tumor_normal_df,
+  aes(
+    x = log_fold_change,
+    y = neg_log_p_val,
+    color = avg_expression
+  )
+) +
+  geom_point(alpha = 0.2) +
+  theme_bw() +
+  xlab("log10 Fold Change") + # Add an x label
```

Comment:
I think this is log2, not log10, but check with limma. And we should specify the direction of the comparison: Tumor/Normal, I believe.

Reply:
Yes! You are right. It's log2.